### PR TITLE
Store the PHPCS and PHP Compatibility scan caches across workflow runs

### PR DESCRIPTION
**Pull request by desrosj**
This adds the needed steps to store the PHPCS caches for coding standards and PHP compatibility scanning across workflow runs.

It uses the date of the previous Monday for the cache key to ensure that the cache is regularly refreshed. As the week progresses, some files will need to be manually scanned for each run as they are updated, but this should will not result in big difference, unless many PHP files are updated over the course of a week.

Trac ticket: https://core.trac.wordpress.org/ticket/49783

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: .github/workflows/php-compatibility.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
       - name: Set up PHP
         uses: shivammathur/setup-php@afefcaf556d98dc7896cca380e181decb609ca44 # v2.10.0
         with:
-          php-version: '7.4'
+          php-version: '8.0'
```
Comment (jrfnl) [inline .github/workflows/php-compatibility.yml:63]: Please do not use PHP 8.0 yet with PHPCS, at least not until PHPCS has been updated to 3.6.1, and preferably not until WPCS 3.0.0 and PHPCompatibility 10.0.0 have been released.

Comment (desrosj) [reply .github/workflows/php-compatibility.yml]: Sorry about that! Was testing this out of curiosity and missed changing this one back in a809d4167f084abd39c2307d448864e60dc3ff46.

**Comment by desrosj** [comment]
Before adding caching in this PR, the [PHPCS scan took just over a minute](https://github.com/WordPress/wordpress-develop/runs/4194300542?check_suite_focus=true#step:10:1). After, it has been reduced by over 95% to less than 3 seconds.

The caching the PHP compatibility scanning resulted in similar reduction. It's not currently being shown in action runs for this PR because I believe there is a bad cache value. That cache will be flushed at some point today when the action runner's date changes into Monday and the key resets. I've tested and verified the improvement locally.

**Review by johnbillion — APPROVED** [review]
LGTM. When this is committed, the svn:ignore prop needs to be updated to include the `.cache` directory too to reduce the chance of somebody accidentally committing it.

**Comment by desrosj** [comment]
Merged in https://core.trac.wordpress.org/changeset/52183.

**Pull request closed without merging**